library/core/class.session.php

```diff
@@ -690,7 +690,7 @@ public function validateTransientKey($foreignKey, $validateUser = true) {
         if (!isset($return)) {
             // Checking the postback here is a kludge, but is absolutely necessary until we can test the ValidatePostBack more.
             $return = ($forceValid && Gdn::request()->isPostBack()) ||
-                (hash_equals($this->_TransientKey, $foreignKey) && !empty($this->_TransientKey));
+                (is_string($foreignKey) && hash_equals((string)$this->_TransientKey, $foreignKey) && !empty($this->_TransientKey));
```

Comment:
I'm assuming some of this was done to deal with a notice or warning.

```suggestion
                (!empty($this->_TransientKey) && is_string($foreignKey) && hash_equals((string)$this->_TransientKey, $foreignKey));
```